From e08b791a06ebe38dbd69c7527fe0ac8920c4f1f2 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Tue, 12 Mar 2024 05:08:19 +0200
Subject: [PATCH] [FSW-402] fs1: init: read die id

This patch adds to bootrom init functionality to read
and process die id and die type.

Change-Id: I494d6f60689d8b8e6f6d6581edb41324138a3e27
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 3098ac5..7c5bdf7 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -12,7 +12,7 @@
 
 uint8_t boot_rom_get_die_id(void)
 {
-	uint32_t die_cpu_id = base_readl(mmPARC_LOCAL_DIE_ID);
+	uint32_t die_cpu_id = base_readl(IO0_HALF0_PSOC_ARC_MNG_LOCAL(mmARC_LOCAL_CONF_DIE_CPU_ID));
 
 	return ((die_cpu_id & ARC_LOCAL_CONF_DIE_CPU_ID_DIE_ID_M)
 		>> ARC_LOCAL_CONF_DIE_CPU_ID_DIE_ID_S);
diff --git a/common/common.c b/common/common.c
index e453b89..75b30b3 100644
--- a/common/common.c
+++ b/common/common.c
@@ -14,4 +14,8 @@
 	uint8_t die_id = boot_rom_get_die_id();
 
 	boot_rom_persistent_dccm_init(die_id);
+
+	boot_rom_global.die_id = die_id;
+	boot_rom_global.die_type = boot_rom_get_die_type();
+
 }
